app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,7 @@
+class RecipesController < ApplicationController
+    def show
+        @recipe = Recipe.find(params[:id])
+        @ingredients = @recipe.ingredients
+        @total_cost = @ingredients.sum(:cost)
```

Comment:
So this method would live in your `recipe` model. 
```ruby
def total_cost
  ingredients.sum(:cost)
end
``` 
Then you would call the `total_cost` method in your `recipes/show.html.erb`